project-fortis-interfaces/src/components/Admin/UserRoles.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import Snackbar from 'material-ui/Snackbar';
+import groupBy from 'lodash/groupBy';
+import differenceBy from 'lodash/differenceBy';
+import { DataGrid } from './DataGrid';
+import { getColumns } from './shared';
+const { Editors, Formatters } = require('react-data-grid-addons');
+const { DropDownEditor } = Editors;
+const { DropDownFormatter } = Formatters;
+
+const roles = ['user', 'admin'];
+
+class UserRoles extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      open: false,
+      message: ''
+    };
+
+    this.getUserRolesColumns = this.getUserRolesColumns.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.handleRemove = this.handleRemove.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.flux.actions.ADMIN.load_users()
+  }
+
+  getUserRolesColumns() {
+    const columnValues = [
+      {editable: true, filterable: true, sortable: true, key: "identifier", name: "Identity"},
+      {editable: false, filterable: true, sortable: true, key: "role", name: "Role", editor: <DropDownEditor options={roles}/>, formatter: <DropDownFormatter options={roles}/>}
+    ];
+
+    return getColumns(columnValues);
+  }
+
+  handleSave(users) {
+    const adminRowsMissingUserRole = this.getAllAdminWithoutUserRole(users);
+    if (adminRowsMissingUserRole.length > 0) this.alertAddUserOnAddAdmin(adminRowsMissingUserRole);
+    this.props.flux.actions.ADMIN.add_users(this.removeIdsFromUsers(users));
+  }
+
+  getAllAdminWithoutUserRole(users) {
+    const usersGroupedByRole = groupBy(users, 'role');
+    return differenceBy(usersGroupedByRole.admin, usersGroupedByRole.user, 'identifier');
+  }
+
+  alertAddUserOnAddAdmin(admin) {
+    const adminIdentities = admin.map(a => a.identifier);
+    this.setState({
+      open: true,
+      message: `Added user role to ${adminIdentities}.`
+    });
+  }
+
+  removeIdsFromUsers(users) {
+    return users.map(user => ({identifier: user.identifier, role: user.role}) );
+  }
+
+  handleRemove(users) {
+    this.props.flux.actions.ADMIN.remove_users(this.removeIdsFromUsers(users), (err, res) => {
+      if (!err) {
+        const usersMissingAdminRole = this.getAllUsersWithoutAdminRole(users);
+        if (usersMissingAdminRole.length > 0) this.alertDeleteAdminOnDeleteUser(usersMissingAdminRole);
+      }
+    });
+  }
+
+  getAllUsersWithoutAdminRole(users) {
+    const usersGroupedByRole = groupBy(users, 'role');
+    return differenceBy(usersGroupedByRole.user, usersGroupedByRole.admin, 'identifier');
+  }
+
+  alertDeleteAdminOnDeleteUser(users) {
+    const adminIdentities = users.map(a => a.identifier);
+    this.setState({
+      open: true,
+      message: `Admin roles for ${adminIdentities} were deleted for any user deletions.`
+    });
+  }
+
+  render() {
+    return (
+      this.getUserRolesColumns().length > 0 ?
```

Comment:
Please use the early-return pattern to make it clearer what the function returns in error cases. Here are two background readings that highlight the benefits of early returns: [1](http://www.codergears.com/Blog/?p=2392), [2](https://arne-mertz.de/2016/12/early-return/).